app/components/hosts/host-detail.ts

```diff
@@ -0,0 +1,108 @@
+import {ClusterHelper} from '../clusters/cluster-helpers';
+import {MockDataProvider} from '../clusters/mock-data-provider-helpers';
+import {ClusterService} from '../rest/clusters';
+import {ServerService} from '../rest/server';
+
+declare var require : any;
+var c3 = require("c3");
+var numeral = require("numeral");
+export class HostDetailController {
+    private self = this;
+    private capacity;
+    private mockHost;
+    private iops;
+    private MockDataProvider;
+    private ClusterHelper;
+    private Id;
+
+    static $inject: Array<string> = [
+        '$q',
+        '$scope',
+        '$location',
+        '$log',
+        '$routeParams',
+        'ClusterService',
+        'ServerService',
+    ];
+
+    constructor(private qService: ng.IQService,
+        private scopeSvc: ng.IScope,
+        private locationSvc: ng.ILocationService,
+        private logSvc: ng.ILogService,
+        private routeParams: ng.route.IRouteParamsService,
```

Comment:
use $prefix for all the angular built-in services
like $q, $scope. $location, $log, $routePrarams